File: frontend/src/interface/student.ts
export interface Student {
  id: number
  name: string
  studentNumber: string
  classId: number
  gender: string
  phoneNumber: string
  highestEducation: string
  disciplinaryActions: number
  disciplinaryPoints: number
  lastOperationTime: Date
}

export function isStudent(arg: any): arg is Student {
  return (
    arg.id !== undefined &&
    arg.name !== undefined &&
    arg.studentNumber !== undefined &&
    arg.classId !== undefined &&
    arg.gender !== undefined &&
    arg.phoneNumber !== undefined &&
    arg.highestEducation !== undefined &&
    arg.disciplinaryActions !== undefined &&
    arg.disciplinaryPoints !== undefined &&
    arg.lastOperationTime !== undefined
  )
}

File: backend/src/main/java/com/webdev/backend/model/Student.java
package com.webdev.backend.model;

import java.sql.Timestamp;

import jakarta.annotation.Generated;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "students")
public class Student {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String name;

	private String studentNumber;

	private Integer classId;

	private String gender;

	private String phoneNumber;

	private String highestEducation;

	private Integer disciplinaryActions;

	private Integer disciplinaryPoints;

	private Timestamp lastOperationTime;

	@ManyToOne
	@JoinColumn(name = "classId", insertable = false, updatable = false)
	private Class classUnit;

	public void setLastOperationTime() {
		this.lastOperationTime = new Timestamp(System.currentTimeMillis());
	}

}